library.c

```diff
@@ -63,18 +63,19 @@ extern zend_class_entry *redis_exception_ce;
 
 extern int le_redis_pconnect;
 
-static zend_llist *
+static ConnectionPool *
 redis_sock_get_connection_pool(RedisSock *redis_sock TSRMLS_DC)
 {
     zend_string *persistent_id = strpprintf(0, "phpredis_%s:%d", ZSTR_VAL(redis_sock->host), redis_sock->port);
```

Comment:
This function causes several `defined but not used` warnings in PHP 5.

```bash
In file included from /home/mike/dev/phpfarm/src/php-5.6.38-debug-nozts/ext/redis/php_redis.h:22:0,
                 from /home/mike/dev/phpfarm/src/php-5.6.38-debug-nozts/ext/redis/cluster_library.c:1:
/home/mike/dev/phpfarm/src/php-5.6.38-debug-nozts/ext/redis/common.h:72:1: warning: ‘zend_strpprintf’ defined but not used [-Wunused-function]
 zend_strpprintf(size_t max_len, const char *format, ...)
```

Since it's only used by library.c the function could be moved there under the existing `#if PHP_MAJOR_VERSION < 7` block at the top.  This isn't a bug it just clutters up the build.

Reply:
Moved `zend_strpprintf` to library.c